### PR TITLE
Implement Using

### PR DESCRIPTION
**Pull request by MrLolthe1st**

### CODE REVIEW

**Comment by MrLolthe1st** [comment]
1732

**Comment by MrLolthe1st** [comment]
#2544 1732

File: ydb/library/yql/core/common_opt/yql_co_pgselect.cpp [hunk 1]
```diff
@@ -1519,6 +1530,63 @@ std::tuple<TVector<ui32>, TExprNode::TListType> BuildJoinGroups(TPositionHandle
                 current = cartesian;
                 continue;
             }
+            if (join->ChildrenSize() > 1 && join->Child(1)->Content() == "using") {
```
Comment (vitstn) [inline ydb/library/yql/core/common_opt/yql_co_pgselect.cpp:1533]: please check that join->Child(1) is an Atom and add YQL_ENSURE for its content

join->ChildrenSize() > 1 looks like YQL_ENSURE too
graph should be valid at this moment

File: ydb/library/yql/sql/pg/pg_sql.cpp [hunk 2]
```diff
@@ -930,27 +928,51 @@ class TConverter : public IPGParseEvents {
...
 
-                                if (op != "cross" && !join->quals) {
-                                    AddError("join_expr: expected quals for non-cross join");
-                                    return nullptr;
-                                }
-
-                                if (op == "cross") {
-                                    oneJoinGroup.push_back(QL(QA(op)));
+                                if (ListLength(join->usingClause) > 0) {
+                                    if (op == "cross") {
```
Comment (vitstn) [inline ydb/library/yql/sql/pg/pg_sql.cpp:932]: I guess op SHOULD be "cross" here

File: ydb/library/yql/core/type_ann/type_ann_core.cpp [hunk 3]
```diff
@@ -4708,7 +4708,19 @@ namespace NTypeAnnImpl {
                     .Build();
             }
         }
-
+        if (input->ChildrenSize() > 2) {
```
Comment (vitstn) [inline ydb/library/yql/core/type_ann/type_ann_core.cpp:4711]: why?

File: ydb/library/yql/core/common_opt/yql_co_pgselect.cpp [hunk 4]
```diff
@@ -924,6 +924,19 @@ TUsedColumns GatherUsedColumns(const TExprNode::TPtr& result, const TExprNode::T
         for (ui32 i = 0; i < groupTuple->ChildrenSize(); ++i) {
             auto join = groupTuple->Child(i);
             auto joinType = join->Child(0)->Content();
+            if (join->ChildrenSize() > 1 && join->Child(1)->Content() == "using") {
```
Comment (vitstn) [inline ydb/library/yql/core/common_opt/yql_co_pgselect.cpp:927]: please validate IsAtom precondition && atom's content & here too (only "using" is supported now)

**Review by vitstn — DISMISSED** [review]

**Review by vitstn — APPROVED** [review]

**Pull request merged**